### ISSUE TITLE
Bug: generateNodesFromDOM removes styles from html

### ISSUE THREAD

**Issue by BanannaLopal** [facebook/lexical#3040]
The function `$generateHtmlFromNodes` generates this code:
`<p class="editor-paragraph"><span class="editor-text-underline">underline</span></p><p class="editor-paragraph"><span class="editor-text-strikethrough">strikethrough</span></p><p class="editor-paragraph"><span>align center</span></p><p class="editor-paragraph"><span>align right</span></p>`

But if you convert this html to a node with the function `$generateNodesFromDOM` you get the following code:
`<p class="editor-paragraph ltr" dir="ltr"><span data-lexical-text="true">underline</span></p><p class="editor-paragraph ltr" dir="ltr"><span data-lexical-text="true">strikethrough</span></p><p class="editor-paragraph ltr" dir="ltr"><span data-lexical-text="true">align center</span></p><p class="editor-paragraph ltr" dir="ltr"><span data-lexical-text="true">align right</span></p>`

Lexical version: 0.4.1

## Steps To Reproduce

1. Use $generateHtmlFromNodes to generate HTML
2. Use $generateNodesFromDOM to parse HTML to NODE

Link to code example: https://codesandbox.io/s/lexical-list-parsing-bug-forked-4u27pu?file=/src/Editor.js

## The current behavior
Underline, strikethrough and alignment styles are missing

## The expected behavior
Styles are not lost

**Comment by BanannaLopal** [comment]
monkey patch for underline style

https://github.com/facebook/lexical/issues/2452#issuecomment-1189672038

**Comment by chalecki** [comment]
I think the `$generateNodesFromDOM` is not a problem and the issue is somewhere else and I might discover something regarding this. 

TL;DR

I've checked the `ParagraphNode` code and it looks like it does not inherit styles (eg. indent) from the HTML.

Correct me if I'm wrong, but the flow looks like this:

First, the `importDOM` method is invoked.

```tsx
static importDOM(): DOMConversionMap | null {
  return {
    p: (node: Node) => ({
      conversion: convertParagraphElement,
      priority: 0,
    }),
  };
}
```

Then `createDOM`

```tsx
createDOM(config: EditorConfig): HTMLElement {
  const dom = document.createElement('p');
  const classNames = getCachedClassNameArray(config.theme, 'paragraph');
  if (classNames !== undefined) {
    const domClassList = dom.classList;
    domClassList.add(...classNames);
  }
  return dom;
}
```

From there, we run `convertParagraphElement` and at this point, we create a new, empty paragraph node that inherits `__format = 0` and `__indent` from `ElementNode` which by default is `0`. The imported node is not passed down this function so all styles (e.g. text alignment, text-indent) are basically lost.

```tsx
function convertParagraphElement(): DOMConversionOutput {
  return {node: $createParagraphNode()};
}
```

So, when parsing this HTML:

```
<p class="PlaygroundEditorTheme__paragraph" dir="ltr" style="padding-inline-start: 40px; text-align: right;"><span>right aligned</span></p>
```

I should get a new `ParagraphNode` with
- indent: 2
- format: 3

**Comment by ivailop7** [comment]
Alignment is now fixed in 0.11 for Heading and Quote nodes, for the remaining HTML - underline, etc. you can override the node and use the example here: https://lexical.dev/docs/concepts/serialization#handling-extended-html-styling
